src/nodes.coffee

```diff
@@ -4006,6 +4001,7 @@ exports.Code = class Code extends Base
       # We never generate named functions, so specify `id` as `null`, which
       # matches the Babel AST for anonymous function expressions/arrow functions
       id: null
+      hasIndentedBody: @body.locationData.first_line > @funcGlyph?.locationData.first_line
```

Comment:
Added explicit `hasIndentedBody` flag to function AST nodes (eg `FunctionExpression`) since otherwise there's no reliable way to detect whether a function has an indented (vs inline) body just based on other existing AST location data

To illustrate that, look at eg:
```
(
  a
) -> b
```
vs:
```
(a) ->
  b
```
Since location data for the `)` and `->` tokens aren't surfaced in the AST, you'd have to try and figure out whether the body is inline by looking at whether the preceding AST element with location data (here, `a`) is on the same line as the beginning of the body's location data (here, `b`)

But in both of these examples, `b` starts on the next line after `a` ends, even though in one case we have an indented function body and in the other we have an inline function body

So it makes sense to explicitly expose that as a flag on the AST since when generating AST we still have access to the location data for `->` (`@funcGlyph`) so can reliably determine it

And it's something that the Prettier plugin wants to be able to determine in order to avoid inlining function bodies that weren't inline in the original source